lib/config.js

```diff
@@ -124,10 +132,47 @@ class HomebaseConfig {
     this.writeToFile()
   }
 
+  // calculate disk usage in hosting directory
+  updateDiskUsage() {
+    return new Promise(resolve => {
+      getFolderSize(this.directory, (err, size) => {
+        if (err) {
+          throw err
+        }
+        // - `size` is in bytes
+        // - when we fully parse the tree our "estimate" is exactly correct
+        // - this will drift over time as blocks come in
+        // - we keep a running estimate so we don't need to parse the full
+        //   storage folder on every block that is downloaded
+        this.diskUsage = size
+        this.estimatedDiskUsage = size
+
+        // if we are under quota, restart all dats
+        // is this the right place to do this?? config seems wrong but when we
+        // update the disk space seems right
```

Comment:
You might move this entire function into `server.js` to solve that issue. That's where the vhost management occurs.